frostbite3/dumper.py

```diff
@@ -43,233 +46,243 @@
+    toc=dbo.readToc(tocPath)          
     if not (toc.get("bundles") or toc.get("chunks")): return #there's nothing to extract (the sb might not even exist)
-
-    sbPath=tocPath[:-3]+"sb"
-    sb=open(sbPath,"rb")
-
+                                      
+    sbPath=tocPath[:-3]+"sb"          
+    sb=open(sbPath,"rb")              
+                                      
     chunkPathToc=os.path.join(outPath,"chunks")
     bundlePath=os.path.join(outPath,"bundles")
     ebxPath=os.path.join(bundlePath,"ebx")
     resPath=os.path.join(bundlePath,"res")
     chunkPath=os.path.join(bundlePath,"chunks")
-
+    
+    hasError=0
     ###read the bundle depending on the four types (+cas+delta, +cas-delta, -cas+delta, -cas-delta) and choose the right function to write the payload
-    if toc.get("cas"):
+    if toc.get("cas"):                
         for tocEntry in toc.get("bundles"): #id offset size, size is redundant
-            if tocEntry.get("base"): continue #Patched bundle. However, use the unpatched bundle because no file was patched at all.
-
-            sb.seek(tocEntry.get("offset"))
-            bundle=dbo.DbObject(sb)
-                    
-            #pick the right function
-            if tocEntry.get("delta"):
-                writePayload=payload.casPatchedBundlePayload
-            else:
-                writePayload=payload.casBundlePayload
-
-            for entry in bundle.get("ebx",list()): #name sha1 size originalSize
-                path=os.path.join(ebxPath,entry.get("name")+".ebx")
-                if writePayload(entry,path,False):
-                    ebx.addEbxGuid(path,ebxPath)
-
-            for entry in bundle.get("res",list()): #name sha1 size originalSize resRid resType resMeta
-                path=os.path.join(resPath,entry.get("name")+".res")
-                writePayload(entry,path,False)
-
-            for entry in bundle.get("chunks",list()): #id sha1 size logicalOffset logicalSize chunkMeta::h32 chunkMeta::meta
-                path=os.path.join(chunkPath,entry.get("id").format()+".chunk")
-                writePayload(entry,path,True)
-
+            try: 
+                if tocEntry.get("base"): continue #Patched bundle. However, use the unpatched bundle because no file was patched at all.
+                                          
+                sb.seek(tocEntry.get("offset"))
+                bundle=dbo.DbObject(sb)   
+                                          
+                #pick the right function  
+                if tocEntry.get("delta"): 
+                    writePayload=payload.casPatchedBundlePayload
+                else:                     
+                    writePayload=payload.casBundlePayload
+                                          
+                for entry in bundle.get("ebx",list()): #name sha1 size originalSize
+                    path=os.path.join(ebxPath,entry.get("name")+".ebx") 
+                    if os.path.isfile(path) or writePayload(entry,path,False):
```

Comment:
Can we somehow check if a file for the dump already exists? I found that if the process breaks, it will overwrite previously dumped files. 

My first idea was to have a log file where, at the end of the loop, the file name/relative path/whatever could be held, and to clear the log file when writing the guid table.